### PR TITLE
Refactor evm registry packages

### PR DESCRIPTION
**Pull request by ferglor**
https://smartcontract-it.atlassian.net/browse/AUTO-4035

In this PR, we're moving the `ocr2keepers/evm20` and `ocr2keepers/evm21` packages to `ocr2keepers/evmregistry/v20` and `ocr2keepers/evmregistry/v21` packages respectively.

The changes look pretty intimidating but I've just renamed and relocated two folders using my IDE, so if CI passes I think we should be happy enough!

The JIRA ticket for this specified that we move to `ocr2keepers/evmregistry20` and `ocr2keepers/evmregistry21`, but I thought having a parent evmregistry package with versioned sub packages might be cleaner? Happy to change it to the layout specified in the JIRA if that's preferred!

### CODE REVIEW

File: core/services/ocr2/plugins/ocr2keeper/util.go [hunk 1]
```diff
@@ -20,9 +20,9 @@ import (
 	"github.com/smartcontractkit/chainlink/v2/core/services/job"
 	"github.com/smartcontractkit/chainlink/v2/core/services/keystore/keys/ethkey"
 	"github.com/smartcontractkit/chainlink/v2/core/services/ocr2/models"
-	kevm20 "github.com/smartcontractkit/chainlink/v2/core/services/ocr2/plugins/ocr2keeper/evm20"
-	kevm21 "github.com/smartcontractkit/chainlink/v2/core/services/ocr2/plugins/ocr2keeper/evm21"
-	kevm21transmit "github.com/smartcontractkit/chainlink/v2/core/services/ocr2/plugins/ocr2keeper/evm21/transmit"
+	evm2 "github.com/smartcontractkit/chainlink/v2/core/services/ocr2/plugins/ocr2keeper/evmregistry/v20"
```
Comment (amirylm) [inline core/services/ocr2/plugins/ocr2keeper/util.go:23]: let's use `kevm2` or `evmregistry20`

File: core/services/ocr2/plugins/ocr2keeper/util.go [hunk 2]
```diff
@@ -20,9 +20,9 @@ import (
...
+	evm3 "github.com/smartcontractkit/chainlink/v2/core/services/ocr2/plugins/ocr2keeper/evmregistry/v21"
```
Comment (amirylm) [inline core/services/ocr2/plugins/ocr2keeper/util.go:24]: let's use `kevm21` or `evmregistry21`

Comment (ferglor) [reply core/services/ocr2/plugins/ocr2keeper/util.go:24]: What's the significance of the k? Keepers?

Comment (amirylm) [reply core/services/ocr2/plugins/ocr2keeper/util.go]: yep, to distinguish `evm` from our keepers/automation evm package. I'd prefer `evmregistry21`

**Review by cmalec — DISMISSED** [review]

File: core/services/ocr2/delegate.go [hunk 3]
```diff
@@ -1148,8 +1148,8 @@ func (d *Delegate) newServicesOCR2Keepers21(
 		PayloadBuilder:               services.PayloadBuilder(),
 		UpkeepProvider:               services.UpkeepProvider(),
 		UpkeepStateUpdater:           services.UpkeepStateStore(),
-		UpkeepTypeGetter:             ocr2keeper21core.GetUpkeepType,
```
Comment (infiloop2) [inline core/services/ocr2/delegate.go:1151]: can we still keep the name ocr2keeper21core as this file is common across both versions

**Review by infiloop2 — APPROVED** [review]

**Pull request merged**